lib/faraday/adapter/http_gem.rb

```diff
@@ -21,10 +29,12 @@ def call(env)
               :headers => env[:request_headers],
               :proxy => proxy_options(env)
             }.merge(socket_options(env)))
-        rescue HTTP::ConnectionError => e
-          raise Faraday::Error::ConnectionFailed, e
-        rescue OpenSSL::SSL::SSLError => e
-          raise Faraday::SSLError, e
+        rescue *HTTP_EXCEPTIONS => err
+          if defined?(OpenSSL) && OpenSSL::SSL::SSLError === err
```

Comment:
It seems a little odd to `rescue` a group of exception classes and then check their class. I think leaving two `rescue` clauses would be simpler and easier to read. Is `OpenSSL` not defined in some cases? That could justify this implementation.